### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by rachidelaid**
> tasks done:
- create `Person` class in `person.rb`.
- create `Student` class in `student.rb` Inherits from `Person`.
- create `Teacher` class in `teacher.rb` Inherits from `Person`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (TedLivist) [inline person.rb:14]: - [x] This should be a private method. Kindly refactor accordingly

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def can_use_services?
+    @parent_permision || of_age?
```
Comment (TedLivist) [inline person.rb]: - [x] Your spelling here is wrong. It should be `parent_permission` (double **'s'**, not one). Kindly fix that

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative('./person')
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom)
+    @classroom = classroom
+    super
+  end
```
Comment (TedLivist) [inline student.rb]: - [x] Your `initialize` method here is missing the other parameters just as we have in the Person `initialize` method: `age`, `name` and `parent_permission`. Kindly add them.

- [x] Pass on these other parameters to the Person class though the `super()` method, like so: `super(age. name, parent_permission)`

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative('./person')
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization)
+    @specialization = specialization
+    super
+  end
```
Comment (TedLivist) [inline teacher.rb]: - [x] You are missing the other parameters as in the Person class. Kindly add them.

- [x] Kindly ensure you pass up these parameters on the inherited **Person** class.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @rachidelaid,

Good job so far!
✔️ Good documentation
✔️ Classes are in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉🍾✔**
Hello @rachidelaid 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**